sfge/src/test/java/com/salesforce/rules/usewithsharingondatabaseoperation/SharingPolicyInheritanceParentTest.java

```diff
@@ -0,0 +1,150 @@
+package com.salesforce.rules.usewithsharingondatabaseoperation;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/** tests when MyClass containing a database operation extends ParentClass */
+public class SharingPolicyInheritanceParentTest extends UseWithSharingOnDatabaseOperationTest {
+
+    // spotless:off
+    private static final String PARENT_CLASS_SOURCE =
+        "public %s class ParentClass {\n"
+            + "void foo() {\n"
+            + "}\n"
+        + "}\n";
+
+    private static final String MY_CLASS_SOURCE =
+        "public %s class " + MY_CLASS + " extends ParentClass {\n"
+            + "void foo(Account a) {\n"
+            + "    %s;\n"
+            + "}\n"
+        + "}\n";
+    // spotless:on
+
+    private static final int SINK_LINE = 3;
+
+    // there are 4 * 4 = 16 possible combinations with two sharing policies
+
+    @MethodSource("provideNoViolationTests")
+    @ParameterizedTest(name = "{displayName}: {0} in class with \"{1}\" with parent \"{2}\"")
+    public void testInheritanceNoViolation(String operation, String policyOne, String policyTwo) {
+        String[] sourceCode = {
+            String.format(PARENT_CLASS_SOURCE, policyOne),
+            String.format(MY_CLASS_SOURCE, policyTwo, operation),
+        };
+        assertNoViolation(RULE, sourceCode);
+    }
+
+    private static Stream<Arguments> provideNoViolationTests() {
+        Stream.Builder<Arguments> sb = Stream.builder();
+        Stream<Arguments> databaseOperations = provideSelectDatabaseOperations();
+
+        // spotless:off
+        String[] noViolationSharingPolicyCombos = {
+            "with sharing      , inherited sharing ",
+            "inherited sharing , inherited sharing ",
+            "without sharing   , inherited sharing ",
+            "                  , inherited sharing ",
+            "with sharing      , with sharing      ",
+            "without sharing   , with sharing      ",
+            "inherited sharing , with sharing      ",
+            "                  , with sharing      ",
+        };
+        // spotless:on
+
+        databaseOperations.forEach(
+                operationArg -> {
+                    for (String sharingPolicyCombo : noViolationSharingPolicyCombos) {
+                        String[] policies = sharingPolicyCombo.split(",");
```

Comment:
Instead of using `.split()` here, at line 100, and at line 139, can you use a 2D array? (i.e.,g `String[][]`?)